client/dashboard/utils/purchase.ts

```diff
@@ -263,6 +290,9 @@ export function getTitleForDisplay( purchase: Purchase ): string {
  * the product type as the subtitle.
  */
 export function getSubtitleForDisplay( purchase: Purchase ): string | null {
+	if ( ! purchase ) {
+		return null;
+	}
```

Comment:
This is not necessary as according to the types, purchase will always be a `Purchase`.